mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxUpdateParentAsLookupListener.java

```diff
@@ -0,0 +1,87 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.store.event;
+
+import java.util.List;
+
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxACL;
+import org.apache.james.mailbox.model.MailboxACL.EntryKey;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.transaction.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MailboxUpdateParentAsLookupListener implements MailboxListener {
+    private static final Logger logger = LoggerFactory.getLogger(MailboxUpdateParentAsLookupListener.class);
+    private MailboxSessionMapperFactory mailboxSessionMapperFactory;
+
+    public MailboxUpdateParentAsLookupListener(MailboxSessionMapperFactory mailboxSessionMapperFactory) {
+        this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
+    }
+    @Override
+    public ListenerType getType() {
+        return ListenerType.EACH_NODE;
+    }
+
+    @Override
+    public ExecutionMode getExecutionMode() {
+        return ExecutionMode.SYNCHRONOUS;
+    }
+
+    @Override
+    public void event(Event event) {
+        if (event instanceof EventFactory.MailboxLookupRightUpdate) {
+            try {
+                EventFactory.MailboxLookupRightUpdate mailboxLookupRightUpdate = (EventFactory.MailboxLookupRightUpdate)event;
```

Comment:
Space between ) and event

Reply:
Are we sure yet this is a **mailboxLookupRightUpdate**? I mean, we don't know anything about that, it should be "aclUpdateEvent"

Reply:
The logic you are doing for every acl Event should be done only when **We are setting the 'l' right**. Overwize it do not make sense....